HackIllinois/ViewControllers/HIScanQRCodeViewController.swift

```diff
@@ -0,0 +1,205 @@
+//
+//  HIScanQRCodeViewController.swift
+//  HackIllinois
+//
+//  Created by HackIllinois Team on 11/9/21.
+//  Copyright © 2021 HackIllinois. All rights reserved.
+//  This file is part of the Hackillinois iOS App.
+//  The Hackillinois iOS App is open source software, released under the University of
+//  Illinois/NCSA Open Source License. You should have received a copy of
+//  this license in a file with the distribution.
+//
+
+import Foundation
+import UIKit
+import AVKit
+import CoreData
+import APIManager
+import HIAPI
+
+class HIScanQRCodeViewController: HIBaseViewController {
+    private var captureSession: AVCaptureSession?
+    private let containerView = HIView {
+        $0.translatesAutoresizingMaskIntoConstraints = false
+        $0.backgroundHIColor = \.baseBackground
+    }
+    private let previewView = HIView {
+        $0.backgroundHIColor = \.baseBackground
+    }
+    private var previewLayer: AVCaptureVideoPreviewLayer?
+    let hapticGenerator = UINotificationFeedbackGenerator()
+    private let pickerView = UIPickerView()
+
+    private var loadFailed = false
+    var respondingToQRCodeFound = true
+
+    private let closeButton = HIButton {
+        $0.tintHIColor = \.action
+        $0.backgroundHIColor = \.clear
+        $0.activeImage = #imageLiteral(resourceName: "MenuClose")
+        $0.baseImage = #imageLiteral(resourceName: "MenuClose")
+    }
+}
+
+// MARK: - UIViewController
+extension HIScanQRCodeViewController {
+    override func loadView() {
+        super.loadView()
+        view.addSubview(containerView)
+        view.bringSubviewToFront(containerView)
+        containerView.constrain(to: view, topInset: 0, bottomInset: 0)
+        containerView.constrain(to: view, trailingInset: 0, leadingInset: 0)
+        containerView.addSubview(previewView)
+        view.addSubview(closeButton)
+        closeButton.addTarget(self, action: #selector(didSelectCloseButton(_:)), for: .touchUpInside)
+        closeButton.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: 3).isActive = true
```

Comment:
Could you make the closeButton a bit larger and farther from the edge so it is easer to tap?